src/Agent.Worker/TaskConditionsEvaluator.cs

```diff
@@ -0,0 +1,50 @@
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker
+{
+    public static class TaskConditionsEvaluator
+    {
+        public static bool AreConditionsSatisfied(IList<TaskCondition> conditions, Variables variables, IExecutionContext executionContext)
```

Comment:
this is only used from stepsrunner. can you move this to be a method on stepsrunner?